Makefile

```diff
@@ -0,0 +1,104 @@
+# Makefile for the bundlechanges library.
+
+ifndef GOPATH
+$(warning You need to set up a GOPATH.)
+endif
+
+PROJECT := github.com/juju/bundlechanges
+PROJECT_DIR := $(shell go list -e -f '{{.Dir}}' $(PROJECT))
+
+ifeq ($(shell uname -p | sed -r 's/.*(x86|armel|armhf).*/golang/'), golang)
+	GO_C := golang
+	INSTALL_FLAGS :=
+else
+	GO_C := gccgo-4.9 gccgo-go
+	INSTALL_FLAGS := -gccgoflags=-static-libgo
+endif
+
+define DEPENDENCIES
+  build-essential
+  bzr
+  git
+  $(GO_C)
+endef
+
+default: build
+
+$(GOPATH)/bin/godeps:
+	go get -v launchpad.net/godeps
+
+# Start of GOPATH-dependent targets. Some targets only make sense -
+# and will only work - when this tree is found on the GOPATH.
+ifeq ($(CURDIR),$(PROJECT_DIR))
+
+build:
+	go build $(PROJECT)/...
+
+check:
+	go test $(PROJECT)/...
+
+install:
+	go install $(INSTALL_FLAGS) -v $(PROJECT)/...
+
+clean:
+	go clean $(PROJECT)/...
+
+else
+
+build:
+	$(error Cannot $@; $(CURDIR) is not on GOPATH)
+
+check:
+	$(error Cannot $@; $(CURDIR) is not on GOPATH)
+
+install:
+	$(error Cannot $@; $(CURDIR) is not on GOPATH)
+
+clean:
+	$(error Cannot $@; $(CURDIR) is not on GOPATH)
+
+endif
+# End of GOPATH-dependent targets.
+
+# Reformat source files.
+format:
+	gofmt -w -l .
+
+# Update the project Go dependencies to the required revision.
+deps: $(GOPATH)/bin/godeps
+	$(GOPATH)/bin/godeps -u dependencies.tsv
+
+# Generate the dependencies file.
+create-deps: $(GOPATH)/bin/godeps
+	godeps -t $(shell go list $(PROJECT)/...) > dependencies.tsv || true
+
+# Install packages required to develop the bundlechanges lib and run tests.
+APT_BASED := $(shell command -v apt-get >/dev/null; echo $$?)
+sysdeps:
```

Comment:
Upon further review, and a test run on jenkins, ignore this request.  It would be nice but our jenkins environment doesn't give sudo privileges to the jenkins user.